src/devices/Lcm1602a1/HD44780.cs

```diff
@@ -0,0 +1,559 @@
+        /// </summary>
+        /// <remarks>
+        /// Timings are based off the original HD44780 specification, which dates back
+        /// to the late 1980s. Modern LCDs can often respond faster. In addition we spend
+        /// time in other code between pulsing in new data, which gives quite a bit of
+        /// headroom over the by-the-book waits.
+        /// 
+        /// At a certain point the benefit is shadowed by time spent on other code. When
+        /// hooked up via 4 data lines there is more room for improvement. In tests on
+        /// one 20x4 LCD times were improved up to a 0.5 modifier.
+        /// 
+        /// There is a busy signal that can be checked that could make this moot, but
+        /// currently we are unable to check the signal fast enough to make gains (or
+        /// even equal) going off hard timings. The busy signal also requires having a
+        /// r/w pin attached. And lastly, as already stated, it takes time to set up
+        /// for another byte of data to be pulsed in.
+        /// </remarks>
+        public double TimingMultiplier { get; set; } = 1.0;
+
+        /// <summary>
+        /// Initializes a new HD44780 LCD controller.
+        /// </summary>
+        /// <param name="registerSelect">The pin that controls the regsiter select.</param>
+        /// <param name="enable">The pin that controls the enable switch.</param>
+        /// <param name="data">Collection of pins holding the data that will be printed on the screen.</param>
+        /// <param name="size">The logical size of the LCD.</param>
+        /// <param name="backlight">The optional pin that controls the backlight of the display.</param>
+        /// <param name="readWrite">The optional pin that controls the read and write switch.</param>
+        /// <param name="controller">The controller to use with the LCD. If not specified, uses the platform default.</param>
+        public HD44780(int registerSelect, int enable, int[] data, Size size, int backlight = -1, int readWrite = -1, IGpioController controller = null)
+        {
+            _rwPin = readWrite;
+            _rsPin = registerSelect;
+            _enablePin = enable;
+            _dataPins = data;
+            _backlight = backlight;
+
+            Size = size;
+
+            _rowOffsets = InitializeRowOffsets(size.Height);
+
+            if (data.Length == 8)
+            {
+                _displayFunction |= DisplayFunction.EightBit;
+            }
+            else if (data.Length != 4)
+            {
+                throw new ArgumentException($"The length of the array given to parameter {nameof(data)} must be 4 or 8");
+            }
+
+            _controller = controller ?? new GpioControllerAdapter(RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
+                ? new GpioController(PinNumberingScheme.Logical, new UnixDriver()) :
+                new GpioController(PinNumberingScheme.Logical, new Windows10Driver()));
```

Comment:
NIT: I see what you are doing here but it isn't super readable. I wonder if it would help if you also add a new line before the RuntimeInformation...

Reply:
Sure

Reply:
Removed this and I allow the default one (which is way faster).